### PR TITLE
Fix GH workflow go and OS versions

### PR DESCRIPTION
**Pull request by pacevedom**
Use same OS version for all steps in workflows, plus bump the go version to the one used to build the container images.

### CODE REVIEW

**Comment by pacevedom** [comment]
cc @dagrayvid

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
         persist-credentials: false
     - uses: actions/setup-go@v2
       with:
-        go-version: '1.16.3'
```
Comment (qbarrand) [inline .github/workflows/e2e.yaml:27]: Would `1.16` work? Then we don't have to update the workflow to follow new go releases.

Comment (pacevedom) [reply .github/workflows/e2e.yaml:27]: It should work, but this was just to keep things aligned with the go version used when building containers. That way we minimize the risk of having differences.

**Pull request merged**